### PR TITLE
tests: remove `Window.vscode()` and related logic

### PR DESCRIPTION
**Pull request by JadenSimon**
## Problem
Writing wrappers for the sole purpose of making an implementation easier to test is not a sustainable approach.

## Solution
* Remove `Window.vscode()` and `FakeWindow`
* Extend `TestWindow` to cover the same scenarios that `FakeWindow` was used for

### Important Notes
This PR uses `Proxy` heavily in order to make "testable" versions of native `vscode` objects. Some things like `showInformationMessage` and `withProgress` do _not_ have an API that can be manipulated programmatically. For these methods, we have to create in-mem representations that can be tested. These currently live in `src/test/shared/vscode/message.ts`.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -92,6 +92,10 @@ export async function activate(context: vscode.ExtensionContext) {
         handleError(error, info.id, defaultMessage)
     })
 
+    if (isCloud9()) {
+        vscode.window.withProgress = wrapWithProgressForCloud9(globals.outputChannel)
+    }
+
```
Comment (JadenSimon) [inline src/extension.ts:98]: This is the only non-test related change. Previously the output channel was wrapped for VSC and C9. Now it's just C9.

File: src/test/globalSetup.test.ts [hunk 2]
```diff
@@ -22,15 +22,15 @@ import { activateExtension } from '../shared/utilities/vsCodeUtils'
 import { FakeExtensionContext, FakeMemento } from './fakeExtensionContext'
 import { TestLogger } from './testLogger'
 import * as testUtil from './testUtil'
-import { createTestWindow } from './shared/vscode/window'
+import { getTestWindow, resetTestWindow } from './shared/vscode/window'
 
 const testReportDir = join(__dirname, '../../../.test-reports')
 const testLogOutput = join(testReportDir, 'testLog.log')
 const globalSandbox = sinon.createSandbox()
+const maxTestDuration = 30_000
```
Comment (JadenSimon) [inline src/test/globalSetup.test.ts:30]: Arbitrary max duration for unit tests. 30 seconds is already very generous but we have some tests that are taking ~7-8 seconds on my machine. Those should probably be moved or optimized.

File: src/extension.ts [hunk 3]
```diff
@@ -317,6 +321,35 @@ function recordToolkitInitialization(activationStartedOn: number, logger?: Logge
     }
 }
 
+/**
+ * Wraps the `vscode.window.withProgress` functionality with functionality that also writes to the output channel.
+ *
+ * Cloud9 does not show a progress notification.
+ */
+function wrapWithProgressForCloud9(channel: vscode.OutputChannel): typeof vscode.window['withProgress'] {
```
Comment (justinmk3) [inline src/extension.ts:329]: should this live in shared/vscode/ or somewhere like that?

**Review by justinmk3 — APPROVED** [review]

File: src/extension.ts (hunk 3 above)
Comment (JadenSimon) [reply src/extension.ts:329]: Depends on how visible we want this behavior/edge-case. Previously it was kind of tucked away so putting it in the extension entry point makes it more visible, maybe too much?

**Pull request merged**